components/datetime/src/pattern/runtime/generic.rs

```diff
@@ -0,0 +1,73 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::{
+    super::{reference, PatternError},
+    super::{GenericPatternItem, PatternItem},
+    Pattern,
+};
+use alloc::vec::Vec;
+use zerovec::{ule::AsULE, ZeroVec};
+
+pub struct GenericPattern<'data> {
+    pub items: ZeroVec<'data, GenericPatternItem>,
+}
+
+impl<'data> GenericPattern<'data> {
+    pub fn combined(self, replacements: Vec<Pattern>) -> Result<Pattern, PatternError> {
```

Comment:
This is all private. Do we put that much effort to document private code?

Reply:
I had to add a bunch of `to_string` to make this example work. It's all right for now, since I'm going to clean up generic patterns and the split between runtime and reference patterns after I move the data provider, but bear in mind that I'm documenting private temporary code :)